### PR TITLE
AND-3745 | Add data class for callback method

### PR DESCRIPTION
**Pull request by pratik-pdc**
[Implemented Data classes for callback method](https://github.com/phonedotcom/pdc-voip-android-sdk/commit/609f1ad30117436c3dfa3794e22fe439651222d9)

 - CallScreenState class been improvised to  new CallEvent class
 - Added MissedCallData and IncomingCallData
 - onCallState changes to onCallEvent
 - Access Modified restricted to package-private for Logger class
 
 
 [Added check for active call and foreground service in Incoming Video Feed](https://github.com/phonedotcom/pdc-voip-android-sdk/commit/a179309d9f2a689563da26ba31afa112482b35e5)

### CODE REVIEW

File: sipservice/build.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
 apply plugin: 'com.android.library'
 apply plugin: "maven-publish"
 apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
```
Comment (hchauhansf) [inline sipservice/build.gradle]: Just for a note -> 'kotlin-android-extensions' is deprecated in new version so we might have to remove it later. for current version its fine.

**Review by hchauhansf — APPROVED** [review]
Looks good to merge 👍

**Review by hchauhansf — APPROVED** [review]

File: sipservice/build.gradle (hunk 1 above)
Comment (pratik-pdc) [reply sipservice/build.gradle]: Good catch! 

Resolved now, replaced with `kotlin-parcelize`

**Review by hchauhansf — APPROVED** [review]
Good to merge 👍

**Pull request merged**